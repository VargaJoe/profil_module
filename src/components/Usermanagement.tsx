import * as React				 	from 'react';
import Title                        from './Title';
import axios                        from 'axios';
import { connect }                  from 'react-redux';

interface ParamList {
    guid: string;
}

interface Params {
    params: ParamList;
}

interface Props {
    pageTitle: string;
    match: Params;
    actionType: boolean; // false - forrgot, true - change password
    userId: number;
    history: any;
}

interface State {
    newPass: string;
    repeaPass: string;
    notEqual: boolean;
    oldPass: string;
}

const DATA = require('../config.json');

export class Usermanagement extends React.Component<Props, State> {
	constructor(prop: Props) {
        super(prop);
        this.state = {
            newPass: '',
            repeaPass: '',
            oldPass: '',
            notEqual: false
        };
        
        this.clickHandler = this.clickHandler.bind(this);
        this.changeHandler = this.changeHandler.bind(this);
        this.blurHandler = this.blurHandler.bind(this);
    }
    
    clickHandler = () => {
        let guid = this.props.match.params.guid;
        let passChange;
        if ( this.props.actionType ) {
            console.log(this.props.userId);
            passChange = {
                UserPsw: this.state.newPass,
                UserId: this.props.userId,
                OldPsw: this.state.oldPass
            };
        } else {
            passChange = {
                UserPsw: this.state.newPass,
                UserId: guid
            };
        }
        axios.post(DATA.odataDomain + '/Root/Sites/Profil(\'ForgottenPassword\')/BisonProfileSetUserPassword', passChange)
        .then(res => {
            console.log(res);
            this.props.history.push('/login');
        });
    }

    blurHandler = () => {
        if ( this.state.newPass !== this.state.repeaPass ) {
            this.setState({
                notEqual: true
            });
        } else {
            this.setState({
                notEqual: false
            });
        }
    }
    
    changeHandler = (e: any) => {
        let name = e.target.name;
        switch (name) {
            case 'NewPasssword':
                this.setState({
                    newPass: e.target.value
                });
                break;
            case 'RepeatPasssword':
                this.setState({
                    repeaPass: e.target.value
                });
                break;
            case 'OldPasssword':
                this.setState({
                    oldPass: e.target.value
                });
                break;
            default:
                break;
        }
	}
	
	public render () {
		return (
			<div className="sn_change_password">
                <Title name={this.props.pageTitle} />
                {this.props.actionType ? 
                    (   
                        <div>
                            <label htmlFor="OldPasssword">Enter old password</label>
                            <input 
                                type="password" 
                                id="OldPasssword"
                                name="OldPasssword"
                                autoComplete="nope"
                                onChange={this.changeHandler}
                                onBlur={this.blurHandler}
                                required={true}
                            />
                        </div>
                    )
                : '' }
                <label htmlFor="NewPasssword">Enter New password</label>
                <input 
                    type="password" 
                    id="NewPasssword"
                    name="NewPasssword"
                    autoComplete="nope"
                    onChange={this.changeHandler}
                    onBlur={this.blurHandler}
                    required={true}
                />
                <label htmlFor="RepeatPasssword">Repeat password</label>
                <input 
                    type="password" 
                    id="RepeatPasssword" 
                    name="RepeatPasssword"
                    autoComplete="nope"
                    onChange={this.changeHandler}
                    onBlur={this.blurHandler}
                    required={true}
                />
                <p>
                    {this.state.notEqual ? 
                        'Passwords are not equal' 
                    : 
                        ''
                    }
                </p>
                <button 
                    className="sn_btn"
                    onClick={this.clickHandler}
                    disabled={this.state.notEqual}
                >
                    Change Password
                </button>
			</div>
		);
	}
}

export const mapStateToProps = (state: any, match: any) => {
    return {
        userId :      state.user.user.Id,
    };
};

export default connect(
    mapStateToProps,
)(Usermanagement as any);