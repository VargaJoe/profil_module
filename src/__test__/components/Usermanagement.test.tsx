import * as React 						  	from 'react';
import Usermanagement ,
		{ UsermanagementComponent }			from '../../components/Usermanagement';
import { 
	configure, 
		shallow,
		mount, 
		ReactWrapper, 
		ShallowWrapper } 					from 'enzyme';
import * as createMockStore					from 'redux-mock-store';
import * as Adapter 					  	from 'enzyme-adapter-react-16';
import toJson 				      			from 'enzyme-to-json';
import { Provider } from 'react-redux';

configure( {adapter: new Adapter()} );

describe('<Usermanagement /> shallow rendering', () => {
	let store;
	let umMount: ReactWrapper<any, any>;
	let umcMount: ReactWrapper<any, any>;
    let umShallow: ShallowWrapper<any, any>;
    let umMount2: ReactWrapper<any, any>;
	let umcMount2: ReactWrapper<any, any>; 
	let umShallow2: ShallowWrapper<any, any>;

	const mockStore = (createMockStore as any)();
	beforeEach( () => {
        store = mockStore({
			sensenet: {
				session: {
					loginState: 'Unauthenticated',
					user: {
						userName: 'Visitor'
					}
				}
            },
			user: {
				user: {
					Id: 12312,
					FullName: 		'adasdasdasd',
					JobTitle: 		'adasdasdasd',
					Email: 			'adasdasdasd',
					Languages: 		'adasdasdasd',
					Phone: 			'adasdasdasd',
					BirthDate: 		'adasdasdasd',
					Education: 		'adasdasdasd',
					Description: 	'adasdasdasd',
				}
			}
        });
        const props = {
            match: {
                params: {
                    guid: 'some GUI - d'
                }
            },
            actionType: false
        };
        const props2 = {
            match: {
                params: {
                    guid: 'some GUI - d'
                }
            },
            actionType: true
        };
		umMount = mount(
			<Provider store={store}>
				<Usermanagement {...props} />
			</Provider>); 
		umcMount = mount(
			<Provider store={store}>
				<Usermanagement  {...props} />
			</Provider>); 
		umShallow = shallow(
			<Provider store={store}>
				<Usermanagement {...props} />
            </Provider>); 
        umMount2 = mount(
			<Provider store={store}>
				<Usermanagement {...props2} />
			</Provider>); 
		umcMount2 = mount(
			<Provider store={store}>
				<Usermanagement  {...props2} />
			</Provider>); 
		umShallow2 = shallow(
			<Provider store={store}>
				<Usermanagement {...props2} />
			</Provider>); 
	});
		
	// test Snapshot 
	it('Match to snapshot', () => {
		expect(toJson(umMount)).toMatchSnapshot();
		expect(toJson(umcMount)).toMatchSnapshot();
    });

    // test Snapshot 
	it('Match to snapshot', () => {
		expect(toJson(umMount2)).toMatchSnapshot();
		expect(toJson(umcMount2)).toMatchSnapshot();
    });
    
    it('test changeHandler ', () => {
		
		const fff = umMount
		.children()
		.children();
		const spy = jest.spyOn(fff.instance(), 'changeHandler'  as any);
		let eventItems = [
			{e: {target: {name: 'NewPasssword', value: 'Loldon Freeman', state: 'newPass'}}},
			{e: {target: {name: 'RepeatPasssword', value: 'Epic Developer', state: 'repeaPass'}}},
            {e: {target: {name: 'OldPasssword', value: 'loldon@mail.ru', state: 'oldPass'}}},
            {e: {target: {name: 'Nothing', value: '', state: ''}}},
		];
		for (let index = 0; index < eventItems.length - 1; index++) {
			const ddd = (fff.instance() as any).changeHandler(eventItems[index].e);
			expect(spy).toBeCalledWith(eventItems[index].e);
			expect(fff.instance().state[eventItems[index].e.target.state]).toBe(eventItems[index].e.target.value);
		}
		
    });
    
    it('test blurHandler ', () => {
		
		const fff = umMount
		.children()
		.children();
        const spy = jest.spyOn(fff.instance(), 'blurHandler'  as any);
        (fff.instance() as any).blurHandler();
		expect(spy).toHaveBeenCalledTimes(1);
    });
    
    it('test clickHandler ', () => {
		
		const fff = umMount
		.children()
		.children();
        const spy = jest.spyOn(fff.instance(), 'clickHandler'  as any);
        (fff.instance() as any).clickHandler();
		expect(spy).toHaveBeenCalledTimes(1);
    });
    
    it('test clickHandler ', () => {
		
		const fff = umMount2
		.children()
		.children();
        const spy = jest.spyOn(fff.instance(), 'clickHandler'  as any);
        (fff.instance() as any).clickHandler();
		expect(spy).toHaveBeenCalledTimes(1);
	});
		 
});
 