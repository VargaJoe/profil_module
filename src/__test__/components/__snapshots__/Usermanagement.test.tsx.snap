// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Usermanagement /> shallow rendering Match to snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(UsermanagementComponent)
    actionType={false}
    match={
      Object {
        "params": Object {
          "guid": "some GUI - d",
        },
      }
    }
  >
    <UsermanagementComponent
      actionType={false}
      match={
        Object {
          "params": Object {
            "guid": "some GUI - d",
          },
        }
      }
      updatePassword={[Function]}
      userId={12312}
      userLogout={[Function]}
    >
      <div
        className="sn_change_password"
      >
        <Title>
          <h2
            className="sn_title"
          />
        </Title>
        <label
          htmlFor="NewPasssword"
        >
          Enter New password
        </label>
        <input
          autoComplete="nope"
          id="NewPasssword"
          name="NewPasssword"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          type="password"
        />
        <label
          htmlFor="RepeatPasssword"
        >
          Repeat password
        </label>
        <input
          autoComplete="nope"
          id="RepeatPasssword"
          name="RepeatPasssword"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          type="password"
        />
        <p />
        <button
          className="sn_btn"
          disabled={false}
          onClick={[Function]}
        >
          Change Password
        </button>
      </div>
    </UsermanagementComponent>
  </Connect(UsermanagementComponent)>
</Provider>
`;

exports[`<Usermanagement /> shallow rendering Match to snapshot 2`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(UsermanagementComponent)
    actionType={false}
    match={
      Object {
        "params": Object {
          "guid": "some GUI - d",
        },
      }
    }
  >
    <UsermanagementComponent
      actionType={false}
      match={
        Object {
          "params": Object {
            "guid": "some GUI - d",
          },
        }
      }
      updatePassword={[Function]}
      userId={12312}
      userLogout={[Function]}
    >
      <div
        className="sn_change_password"
      >
        <Title>
          <h2
            className="sn_title"
          />
        </Title>
        <label
          htmlFor="NewPasssword"
        >
          Enter New password
        </label>
        <input
          autoComplete="nope"
          id="NewPasssword"
          name="NewPasssword"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          type="password"
        />
        <label
          htmlFor="RepeatPasssword"
        >
          Repeat password
        </label>
        <input
          autoComplete="nope"
          id="RepeatPasssword"
          name="RepeatPasssword"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          type="password"
        />
        <p />
        <button
          className="sn_btn"
          disabled={false}
          onClick={[Function]}
        >
          Change Password
        </button>
      </div>
    </UsermanagementComponent>
  </Connect(UsermanagementComponent)>
</Provider>
`;

exports[`<Usermanagement /> shallow rendering Match to snapshot 3`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(UsermanagementComponent)
    actionType={true}
    match={
      Object {
        "params": Object {
          "guid": "some GUI - d",
        },
      }
    }
  >
    <UsermanagementComponent
      actionType={true}
      match={
        Object {
          "params": Object {
            "guid": "some GUI - d",
          },
        }
      }
      updatePassword={[Function]}
      userId={12312}
      userLogout={[Function]}
    >
      <div
        className="sn_change_password"
      >
        <Title>
          <h2
            className="sn_title"
          />
        </Title>
        <div>
          <label
            htmlFor="OldPasssword"
          >
            Enter old password
          </label>
          <input
            autoComplete="nope"
            id="OldPasssword"
            name="OldPasssword"
            onBlur={[Function]}
            onChange={[Function]}
            required={true}
            type="password"
          />
        </div>
        <label
          htmlFor="NewPasssword"
        >
          Enter New password
        </label>
        <input
          autoComplete="nope"
          id="NewPasssword"
          name="NewPasssword"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          type="password"
        />
        <label
          htmlFor="RepeatPasssword"
        >
          Repeat password
        </label>
        <input
          autoComplete="nope"
          id="RepeatPasssword"
          name="RepeatPasssword"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          type="password"
        />
        <p />
        <button
          className="sn_btn"
          disabled={false}
          onClick={[Function]}
        >
          Change Password
        </button>
      </div>
    </UsermanagementComponent>
  </Connect(UsermanagementComponent)>
</Provider>
`;

exports[`<Usermanagement /> shallow rendering Match to snapshot 4`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(UsermanagementComponent)
    actionType={true}
    match={
      Object {
        "params": Object {
          "guid": "some GUI - d",
        },
      }
    }
  >
    <UsermanagementComponent
      actionType={true}
      match={
        Object {
          "params": Object {
            "guid": "some GUI - d",
          },
        }
      }
      updatePassword={[Function]}
      userId={12312}
      userLogout={[Function]}
    >
      <div
        className="sn_change_password"
      >
        <Title>
          <h2
            className="sn_title"
          />
        </Title>
        <div>
          <label
            htmlFor="OldPasssword"
          >
            Enter old password
          </label>
          <input
            autoComplete="nope"
            id="OldPasssword"
            name="OldPasssword"
            onBlur={[Function]}
            onChange={[Function]}
            required={true}
            type="password"
          />
        </div>
        <label
          htmlFor="NewPasssword"
        >
          Enter New password
        </label>
        <input
          autoComplete="nope"
          id="NewPasssword"
          name="NewPasssword"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          type="password"
        />
        <label
          htmlFor="RepeatPasssword"
        >
          Repeat password
        </label>
        <input
          autoComplete="nope"
          id="RepeatPasssword"
          name="RepeatPasssword"
          onBlur={[Function]}
          onChange={[Function]}
          required={true}
          type="password"
        />
        <p />
        <button
          className="sn_btn"
          disabled={false}
          onClick={[Function]}
        >
          Change Password
        </button>
      </div>
    </UsermanagementComponent>
  </Connect(UsermanagementComponent)>
</Provider>
`;
